### PR TITLE
Plugin arguments defaulting + moving plugin arguments to each corresponding plugin #925

### PR DESCRIPTION
**Pull request by spike-liu**
Please feel free to review.

### CODE REVIEW

**Comment by spike-liu** [comment]
this commit resolves #925

**Review by damemi — COMMENTED** [review]
/ok-to-test
thanks for opening this @spike-liu

**Comment by spike-liu** [comment]
> /ok-to-test thanks for opening this @spike-liu

It is my pleasure:)

File: hack/lib/generator-help.sh [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+     find . -type f -name \*.go -not -path "./vendor/*" -print0  \
+     | xargs -0 grep --color=never -l "$@" \
+     | xargs -n1 dirname \
+     | LC_ALL=C sort -u \
+     | sed "s|.|${PRJ_PREFIX}|" \
+     )
+
+   IFS=",";
+   printf '%s' "${array[*]}";
+}
```
Comment (damemi) [inline hack/lib/generator-help.sh:27]: nit: newline

File: hack/update-generated-conversions.sh [hunk 2]
```diff
@@ -5,5 +5,5 @@ go build -o "${OS_OUTPUT_BINPATH}/conversion-gen" "k8s.io/code-generator/cmd/con
 
 ${OS_OUTPUT_BINPATH}/conversion-gen \
 		--go-header-file "hack/boilerplate/boilerplate.go.txt" \
-		--input-dirs "${PRJ_PREFIX}/pkg/apis/componentconfig/v1alpha1,${PRJ_PREFIX}/pkg/api/v1alpha1" \
+		--input-dirs "${PRJ_PREFIX}/pkg/apis/componentconfig/v1alpha1,${PRJ_PREFIX}/pkg/api/v1alpha1,${PRJ_PREFIX}/pkg/framework/plugins/removefailedpods,${PRJ_PREFIX}/pkg/framework/plugins/nodeutilization,${PRJ_PREFIX}/pkg/framework/plugins/podlifetime,${PRJ_PREFIX}/pkg/framework/plugins/removeduplicates,${PRJ_PREFIX}/pkg/framework/plugins/removepodshavingtoomanyrestarts,${PRJ_PREFIX}/pkg/framework/plugins/removepodsviolatinginterpodantiaffinity,${PRJ_PREFIX}/pkg/framework/plugins/removepodsviolatingnodeaffinity,${PRJ_PREFIX}/pkg/framework/plugins/removepodsviolatingnodetaints,${PRJ_PREFIX}/pkg/framework/plugins/removepodsviolatingtopologyspreadconstraint" \
```
Comment (damemi) [inline hack/update-generated-conversions.sh:8]: could we also do something like with deep-copies to not need to list every directory?

File: hack/lib/generator-help.sh (hunk 1 above)
Comment (spike-liu) [reply hack/lib/generator-help.sh]: fixed

File: hack/update-generated-conversions.sh (hunk 2 above)
Comment (spike-liu) [reply hack/update-generated-conversions.sh]: sure. codes have been committed, verified and pushed for conversions & defaulters.

**Comment by damemi** [comment]
@spike-liu can you remove the merge commit and sync this with `git rebase` instead? Thanks

**Comment by spike-liu** [comment]
> @spike-liu can you remove the merge commit and sync this with `git rebase` instead? Thanks

My bad. Fixed and pushed.

Thanks for your review, Mike. Please feel free to let me know if there are anything inappropriate.

**Comment by damemi** [comment]
@spike-liu thanks, it looks like you just need to run `hack/update-vendor.sh` based on the verify-master CI failure

**Comment by spike-liu** [comment]
> @spike-liu thanks, it looks like you just need to run `hack/update-vendor.sh` based on the verify-master CI failure

Appreciate your instruction, Mike.

Build error has been fixed.

**Review by damemi — COMMENTED** [review]
/approve
looks pretty straightforward to me, thanks @spike-liu!
@ingvagabund want to take a second look for lgtm?

**Comment by spike-liu** [comment]
> /approve looks pretty straightforward to me, thanks @spike-liu! @ingvagabund want to take a second look for lgtm?

The pleasure is mine, Mike:)

**Comment by spike-liu** [comment]
> /lgtm This is a good start. @spike-liu thank you for implementing this. We can start building on top of it

Really glad to be a part of the community, Jan:)

**Comment by a7i** [comment]
Thank you @spike-liu 

Ideally, we would squash commits, as opposed to 24 commit showing up on master.

**Pull request merged**